models/bpf.go

```diff
@@ -0,0 +1,135 @@
+// Copyright Contributors to the L3AF Project.
+// SPDX-License-Identifier: Apache-2.0
+
+package models
+
+import (
+	"github.com/cilium/ebpf"
+	"github.com/l3af-project/l3afd/config"
+)
+
+// BPF defines run time details for BPFProgram.
+type BPF interface {
+	// Name returns the BPFProgram Package Name
+	Name() string
+
+	// Artifact returns the BPFProgram Artifact
+	Artifact() string
+
+	// MapName returns the map name of the BPFProgram
+	MapName() string
+
+	// ProgId returns the program ID of the BPFProgram
+	ProgId() int
+
+	// SeqId return the sequence ID of the BPFProgram
+	SeqId() int
+
+	// AdminStatus return the admin status of the BPFProgram
+	AdminStatus() string
+
+	// Version returns the version of the BPFProgram
+	Version() string
+
+	// RestartCount returns the restart count
+	GetRestartCount() int
+
+	// GetProgram return the underlying BPFProgram
+	GetProgram() *BPFProgram
+
+	// GetMapNamePath returns the MapNamePath
+	GetMapNamePath() string
+
+	// GetPrevMapNamePath returns the PrevMapNamePath
+	GetPrevMapNamePath() string
+
+	// AddToRestartCount increases the RestartCount by the passed value
+	AddToRestartCount(value int)
+
+	// UpdateAdminStatus updates the admin status of the BPFProgram to rhe passed string
+	UpdateAdminStatus(value string)
+
+	// UpdatesProgram updates the underlying BPFProgram to the passed BPFProgram struct
+	UpdateProgram(value *BPFProgram)
+
+	// UpdatePrevMapNamePath updates the PrevMapNamePath to the passed string.
```

Comment:
nit: I'm noticing that some comments end in a period (like this one), and some don't.